pkg/image/registryclient/client.go

```diff
@@ -352,19 +421,43 @@ func (c *Context) repositoryTransport(t http.RoundTripper, registry *url.URL, re
 	return c.cachedTransport(t, c.scopes(repoName))
 }
 
+func (c *Context) reposForRef(ctx context.Context, locator imagereference.DockerImageReference, sources []imagereference.DockerImageReference) error {
+	cachedRepos := make(map[imagereference.DockerImageReference]distribution.Repository)
+	for _, altRef := range sources {
+		altRepo, err := c.Repository(ctx, altRef.DockerClientDefaults().RegistryURL(), altRef.RepositoryName(), true)
```

Comment:
No, a repository for the next source shouldn't be created until it is proved that the current source cannot provide the image.

Reply:
ok, I can update that, that was my original assumption but somehow in conversations I was led to believe there was a use-case for collecting possible Repositories, choosing from them - again, I don't know of that use-case just that it possibly exists - @ricardomaraschini also let me know that this is not something image-registry would need.